### PR TITLE
Improved Docker support (Closes #375) + bumped some dependencies

### LINKED ISSUE #375: Improve Docker support + documentation
### Problem

Problem*s*:

- Our documentation on how and why to use Docker is lacking
- Our current documentation is confusing, and recommends using `grunt dev`, but `grunt dev` doesn't work with Docker
- We have no grunt commands for working with Docker
- Updating dependencies and running tests must be done in Docker if Docker is used
- `package-lock.json` will be different depending on whether it's run in Docker or not because of a dependency we have and because `npm` is crap

### Solution

- Review and incorporate the suggestions made by @sandrina-p in #367 PR, but fix some of the issues with that PR (e.g. linking to Mac-specific Docker docs, usage and naming of npm script commands, etc.)
- ~~Make `grunt dev` check for existence of `docker` command, and run itself in Docker if the `docker` command exists. Add an environment variable override to prevent this behavior, and corresponding documentation to both our docs and to the terminal output produced by `grunt dev`~~ Instead, just create a `grunt docker` command that passes the rest of the arguments as comments through Docker.
- ~~Create corresponding grunt commands to install/update dependencies, and to run tests~~ see edit above
- Make the timezone be based on host system's time, not `US/Pacific`

### PR DESCRIPTION
**Pull request by taoeffect**
+ Greatly improved docker support & docs (#375)
+ Updated dependencies: sqlite3, grunt-exec, grunt-sass
+ Fixed [stylelint errors](https://github.com/okTurtles/group-income-simple/pull/370#issuecomment-381438966) that I encountered in PR #370

EDIT:

+ Also, this might close issue #378 (running two GIS instances at the same time)

### How to run two GIS instances

You have to have **two** cloned directories of `group-income-simple`, and both of them have to have the changes in this PR.

Then in one terminal window:

```
$[terminal-1] cd path/to/first/group-income-simple
$[terminal-1] grunt dev    # or npm run docker
```

Then in another terminal window:

```
$[terminal-2] cd path/to/second/group-income-simple
$[terminal-2] PORT_SHIFT=1 grunt dev    # or PORT_SHIFT=1 npm run docker
```

Now in one browser window you visit http://localhost:8000, and in another window (either an incognito window, or a different browser, or a different firefox container tab), visit http://localhost:8001

### CODE REVIEW

**Comment by taoeffect** [comment]
Note: I just pushed some changes to this PR to potentially close issue #378 as well (ability to run two different GIS instances), as it required updating the docker stuff to in order to do that, I decided it made sense to do both in 1 PR.

File: .Gruntfile.babel.js [hunk 1]
```diff
@@ -117,7 +117,7 @@ module.exports = (grunt) => {
       },
       standard: './node_modules/.bin/standard "**/*.{js,vue}"',
       standardgrunt: './node_modules/.bin/standard .Gruntfile.babel.js Gruntfile.js',
-      stylelint: './node_modules/.bin/stylelint "frontend/simple/**/*.{css,scss,vue}"',
+      stylelint: './node_modules/.bin/stylelint "frontend/simple/**/*.{css,scss,vue}, !frontend/simple/js/ignored/**"',
```
Comment (sandrina-p) [inline .Gruntfile.babel.js]: What's this path? I think this path only exists at your machine because that path doesn't exist remotely nor my machine

**Review by sandrina-p — CHANGES_REQUESTED** [review]
On my machine it still throws the same error when I run `PORT_SHIFT=1 grunt dev`
```
Running "connect:dev" (connect) task
Started connect web server on http://localhost:8001

Running "backend:relaunch" task
backend: forking...

Running "watch" task
Waiting...
Fatal error: Port 35729 is already in use by another process.
```

**Comment by taoeffect** [comment]
@sandrina-p My latest changes should fix all the issues you encountered (hopefully) :)

**Review by sandrina-p — APPROVED** [review]
Everything works fine now! 🎉

**Pull request merged**